From 1bc68127d1531ed519cb839844febaecb2a3f6d0 Mon Sep 17 00:00:00 2001
From: Andrew Jones <drjones@redhat.com>
Date: Sat, 9 Jan 2021 22:19:28 -0500
Subject: hw/arm/virt: Add 8.4 Machine type

RH-Author: Andrew Jones <drjones@redhat.com>
Message-id: <20210109221928.31407-3-drjones@redhat.com>
Patchwork-id: 100548
O-Subject: [RHEL-AV-8.4.0 qemu-kvm PATCH v2 2/2] hw/arm/virt: Add 8.4 Machine type
Bugzilla: 1895276
RH-Acked-by: Gavin Shan <gshan@redhat.com>
RH-Acked-by: Miroslav Rezanina <mrezanin@redhat.com>
RH-Acked-by: Auger Eric <eric.auger@redhat.com>
RH-Acked-by: Thomas Huth <thuth@redhat.com>

8.4 isn't much different than 8.3, except it adds the steal-time
feature and enables it by default.

Signed-off-by: Andrew Jones <drjones@redhat.com>
Signed-off-by: Danilo C. L. de Paula <ddepaula@redhat.com>
---
 hw/arm/virt.c | 14 ++++++++++++--
 1 file changed, 12 insertions(+), 2 deletions(-)

diff --git a/hw/arm/virt.c b/hw/arm/virt.c
index 530072fce0..208c360342 100644
--- a/hw/arm/virt.c
+++ b/hw/arm/virt.c
@@ -2894,11 +2894,21 @@ static void rhel_machine_init(void)
 }
 type_init(rhel_machine_init);
 
-static void rhel830_virt_options(MachineClass *mc)
+static void rhel840_virt_options(MachineClass *mc)
 {
     compat_props_add(mc->compat_props, arm_rhel_compat, arm_rhel_compat_len);
 }
-DEFINE_RHEL_MACHINE_AS_LATEST(8, 3, 0)
+DEFINE_RHEL_MACHINE_AS_LATEST(8, 4, 0)
+
+static void rhel830_virt_options(MachineClass *mc)
+{
+    VirtMachineClass *vmc = VIRT_MACHINE_CLASS(OBJECT_CLASS(mc));
+
+    rhel840_virt_options(mc);
+    compat_props_add(mc->compat_props, hw_compat_rhel_8_3, hw_compat_rhel_8_3_len);
+    vmc->no_kvm_steal_time = true;
+}
+DEFINE_RHEL_MACHINE(8, 3, 0)
 
 static void rhel820_virt_options(MachineClass *mc)
 {
-- 
2.18.4

